crypto/hash.go

```diff
@@ -52,8 +58,11 @@ func HashBytes(data []byte) Hash {
 
 // HashObject takes an object as input, encodes it using the encoding package,
 // and then hashes the result.
-func HashObject(obj interface{}) Hash {
-	return HashBytes(encoding.Marshal(obj))
+func HashObject(obj interface{}) (hash Hash) {
+	h := NewHash()
+	encoding.NewEncoder(h).Encode(obj)
+	h.Sum(hash[:0])
+	return
```

Comment:
Ditto here. The test is more for ensuring compatibility if we ever make future changes.